### PR TITLE
Create deploy acnt example

### LINKED ISSUE #251: Fix DeployAccount Example using RPCv2 instead of gateway

### LINKED ISSUE #254: Create an AddDeployAccount Test

### PR DESCRIPTION
**Pull request by rianhughes**
Fixes: https://github.com/NethermindEth/starknet.go/issues/254
Fixes: https://github.com/NethermindEth/starknet.go/issues/251

### CODE REVIEW

File: examples/deployAccount/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+This example uses a pre-existing contract on the goerli network to deploy a new account contract. To successfuly run this example, you will need: 1) a goerli endpoint, and 2) to fund the precomputed address.
```
Comment (cicr99) [inline examples/deployAccount/README.md:2]: typo on _successfuly -> successful**l**y_

File: examples/deployAccount/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+Steps:
+1. Rename the ".env.template" file to ".env.testnet"
+2. Uncomment, and assign your testnet endpoint to the "INTEGRATION_BASE" variable
+3. Execute `go mod tidy` (make sure you are in the deployAccount folder)
+4. Execute `go run main.go`
+5. Fund the precomputed address using a starknet faucet, eg https://faucet.goerli.starknet.io/
+6. Press any key, then enter
+
+At this point your account should be deployed on testnet, and you can use a block explorer like Voyager to view your transaction using the transaction hash.
```
Comment (cicr99) [inline examples/deployAccount/README.md:12]: It would be nice to add a link for Voyager here as well

File: rpc/types_transaction.go [hunk 3]
```diff
@@ -258,6 +258,17 @@ func (v *TransactionVersion) BigInt() (*big.Int, error) {
 	}
 }
 
+func (v *TransactionVersion) Felt() (*felt.Felt, error) {
+	switch *v {
+	case TransactionV0:
+		return &felt.Zero, nil
+	case TransactionV1:
+		return new(felt.Felt).SetUint64(1), nil
+	default:
```
Comment (cicr99) [inline rpc/types_transaction.go:267]: PR #294 adds `TransactionV2`, let's merge that one first and add it here as well, so this example can go in the rpcv03 release

Comment (cicr99) [reply rpc/types_transaction.go:267]: Other than this, I didn't see anything else in this pr that was dependent on rpcv03 changes. Will have to review again once the other prs are merged

**Comment by rianhughes** [comment]
I've recreated this PR here https://github.com/NethermindEth/starknet.go/pull/349 so I don't have to solve two months worth of merge conflicts

**Comment by rianhughes** [comment]
see https://github.com/NethermindEth/starknet.go/pull/349

**Pull request closed without merging**